src/Installer/ContentBlockInstaller.php

```diff
@@ -43,6 +46,21 @@ public function __construct(
 
         $webDir = $composer->getPackage()->getExtra()['typo3/cms']['web-dir'] ?? $defaultWebDir;
         $this->cbsDir = $webDir . DIRECTORY_SEPARATOR . Constants::BASEPATH;
+
+        // make cbsDir absolute
+        if ($rootPackageName !== 'typo3/cms') {
```

Comment:
Is this really necessary? I think that is just the same as a simple `$this->cbsDir = $this->filesystem->normalizePath(realpath($webDir) . DIRECTORY_SEPARATOR . Constants::BASEPATH);` which would get rid of the external dependency.

Reply:
I don't know what the the lines 42-45 do. "TYPO3 in core-dev (git) mode"
Therefore I wanted my changes only for the "normal" case. But it seems, that the `$pluginConfig->get('root-dir')` already contains the "public" element, so I remove the resulting "public/public/".

This can and should be simplyfied and adopted to the extension.

The solution has it's source from `TYPO3\CMS\Core\Composer\CliEntryPoint` where an absolute path is needed too.

Reply:
The "core-dev" mode refers to when TYPO3 is checked out from git for developing on the core. That mode can be identified by the root package name.
This will need the same absolute path fix for composer 2.2 so we do not have to discriminate.

Could you try my suggestion at line 48 and remove lines 50-63?

Reply:
Yes, generates absolute path.